### PR TITLE
[OptApp] Simp Control Allow not removing materials

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
As in the title

**🆕 Changelog**
- Changed the `SimpControl` to disallow removing materials
- Changed the tests accordingly.

### CODE REVIEW

File: applications/OptimizationApplication/tests/control/material/test_simp_control.py [hunk 1]
```diff
@@ -134,6 +138,10 @@ def test_AdaptiveBeta(self):
         parameters = Kratos.Parameters("""{
             "controlled_model_part_names": ["shell"],
             "list_of_materials": [
+                {
+                    "density": 0.0,
+                    "young_modulus": 0.0
```
Comment (Igarizza) [inline applications/OptimizationApplication/tests/control/material/test_simp_control.py:143]: add test please without zeros

Comment (sunethwarna) [reply applications/OptimizationApplication/tests/control/material/test_simp_control.py:143]: altry :)

Comment (sunethwarna) [reply applications/OptimizationApplication/tests/control/material/test_simp_control.py:143]: done :)

**Review by Igarizza — DISMISSED** [review]
Thnx!

**Comment by sunethwarna** [comment]
ping @Igarizza

**Review by Igarizza — DISMISSED** [review]
thnx @sunethwarna

**Review by Igarizza — DISMISSED** [review]

**Review by Igarizza — DISMISSED** [review]

**Review by Igarizza — DISMISSED** [review]

**Review by Igarizza — APPROVED** [review]

**Pull request merged**